Dans-Game/controller.rb

```
@@ -0,0 +1,68 @@
+require_relative 'view'
+require_relative 'model'
+
+class Controller
+  include View
+
+  def initialize
+    @beerList = List.new
+  end
+
+  def add_beer
+    @beerList.add_beer Print::serialize_add_beer
+  end
+
+  def edit_beer
+    id = Print::get_input("\nEnter the id of the beer you want to EDIT:").to_i
+    b = @beerList.find_beer id
+    fail "No beer matching id #{id}" unless b
```

Both calls to `find_beer` fail when they don't get a result. Should `find_beer` be responsible for failing when it doesn't find one?

---

You're right, `find_beer` should probably handle its own failure so there's a single point of failure instead of having to repeat this logic here and in `edit_beer` on line 41 of the model.

I guess the only reason to return nil and check outside `find_beer` would be if I wanted to do some other error handling instead of failing right away, which I didn't do anyway haha.